src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationOptions.cs

```diff
@@ -0,0 +1,217 @@
+        }
+
+        /// <summary>
+        /// Gets the sets the URI the middleware uses to obtain the public key for
+        /// validating tokens if <see cref="ValidateTokens"/> is <see langword="true"/>.
+        /// </summary>
+        public string JwksEndpoint { get; internal set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the Authority to use when making OpenId Connect calls.
+        /// </summary>
+        public string Authority { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets the sets the URI the middleware uses to obtain the OpenId Connect configuration.
+        /// </summary>
+        public string MetadataAddress { get; internal set; } = string.Empty;
+
+        /// <summary>
+        /// Responsible for retrieving, caching, and refreshing the configuration from metadata.
+        /// If not provided, then one will be created using the MetadataAddress and Backchannel properties.
+        /// </summary>
+        public IConfigurationManager<OpenIdConnectConfiguration>? ConfigurationManager { get; set; }
+
+        /// <summary>
+        /// Sets the target online environment to either development, stage or production.
+        /// </summary>
+        public SuperOfficeAuthenticationEnvironment Environment
+        {
+            get
+            {
+                return _environment;
+            }
+
+            set
+            {
+                _environment = value;
+                UpdateEndpoints();
+            }
+        }
+
+        /// <summary>
+        /// When true user claims will include SuperOffice function rights.
+        /// </summary>
+        public bool IncludeFunctionalRightsAsClaims { get; set; }
+
+        /// <summary>
+        /// When true user claims will include SuperOffice id_token claims.
+        /// </summary>
+        public bool IncludeIdTokenAsClaims { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to validate tokens using SuperOffice's public key.
+        /// </summary>
+        public bool ValidateTokens { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether HTTPS is required.
+        /// </summary>
+        public bool RequireHttpsMetadata { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the default period of time to cache the SuperOffice public key(s)
+        /// retrieved from the endpoint specified by <see cref="JwksEndpoint"/>.
+        /// </summary>
+        /// <remarks>
+        /// The default public key cache lifetime is 15 minutes.
+        /// </remarks>
+        public TimeSpan JwksCacheLifetime { get; set; } = TimeSpan.FromMinutes(15);
+
+        /// <summary>
+        /// 1 day is the default time interval that afterwards, <see cref="ConfigurationManager" /> will obtain new configuration.
+        /// </summary>
+        public TimeSpan AutomaticRefreshInterval { get; set; } = TimeSpan.FromDays(1);
```

Comment:
I suppose they are taken from the OIDC handler, but I think we should remove them, as they mostly make sense in a generic handler. Instead, use whatever values you think are appropriate, based on SuperOffice's key rotation policy.

Reply:
I did. Yeah, I've cleaned this up.